e2e/kubectl.e2e.spec.ts

```diff
@@ -0,0 +1,117 @@
+import path from 'path';
+import {
+  ElectronApplication, BrowserContext, _electron, Page, Locator
+} from 'playwright';
+import { test, expect } from '@playwright/test';
+import { TestUtils } from './utils/TestUtils';
+import * as tools from './utils/ToolsUtils';
+
+let page: Page;
+const defaultReportFolder = path.join(__dirname, 'reports/');
+
+test.describe.serial('Rancher Desktop - K8s Deploy Test', () => {
+  let mainTitle: Locator;
+  let utils: TestUtils;
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  const mainTitleSelector = '[data-test="mainTitle"]';
+
+  test.beforeAll(async() => {
+    utils = new TestUtils();
+    utils.createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: `${ defaultReportFolder }pw-trace.zip` });
+    await electronApp.close();
+  });
+
+  test('should load Rancher Desktop App', async() => {
+    mainTitle = page.locator(mainTitleSelector);
+
+    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+  });
+
+  test('should start loading the background services', async() => {
+    const progressBarSelector = page.locator('.progress');
+
+    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
+    await expect(progressBarSelector).toBeHidden();
+  });
+
+  test('should run Kubernetes on Rancher Desktop (kubectl)', async() => {
+    // CI take longer than local env to start up
+    // giving it some time to proper start the background services
+    if (process.env.CI) {
+      console.log('Waiting for services - CI take a while to load...');
+      await utils.delay(200_000);
```

Comment:
Do we not have anything concrete to wait for? Since this test group is serial, the progress bar has already went away.

Having a hard coded delay is always a bad smell…

Reply:
Yep.
I need to figure it out a better way to avoid this. I'll try some stuff and see how it goes after the fixes on Linux.

Reply:
I tried to use a different approach, like: `kubectl wait --for=condition=Ready pods --all -n kube-image` in order to wait until the RD pods are Ready, but the thing is: If the k8s API is not ready, it will fail right away, running any other kubectl commands. 
Error:
```bash
I1222 14:52:18.791530   12616 versioner.go:58] Get https://127.0.0.1:6443/version?timeout=5s: dial tcp 127.0.0.1:6443: connect: connection refused
The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?
```

That's the reason that I've explicit included the delay, if is CI only. Locally it start right away, but in CI takes couple more seconds to start....

Reply:
In that case, it should be polling with a timeout, instead of a hard-coded delay.  Think of it as: if Microsoft decided to make the runners super-fast tomorrow, we shouldn't have to wait for 3 minutes even though it was ready within one second.